tests/Unit/Database/DatabaseQueryStatementTest.php

```diff
@@ -25,127 +23,111 @@ final class DatabaseQueryStatementTest extends TestCase
+            DatabaseDialect::SQLITE,
             'CREATE TABLE Migration (id INTEGER PRIMARY KEY AUTOINCREMENT, name VARCHAR(255) NOT NULL);',
         ];
     }
 
     #[Test]
     #[DataProvider('provide_fk_create_table_database_drivers')]
-    public function it_can_create_a_foreign_key_constraint(DatabaseDriver $driver, string $validSql): void
+    public function it_can_create_a_foreign_key_constraint(DatabaseDialect $dialect, string $validSql): void
     {
         $statement = (new CreateTableStatement('Book'))
             ->primary()
             ->belongsTo('Book.author_id', 'Author.id', OnDelete::CASCADE)
             ->varchar('name')
-            ->compile($driver->dialect());
+            ->compile($dialect);
 
         $this->assertSame($validSql, $statement);
     }
 
     public static function provide_fk_create_table_database_drivers(): Generator
     {
         yield 'mysql' => [
-            new MySqlDriver(),
+            DatabaseDialect::MYSQL,
             'CREATE TABLE Book (id INTEGER PRIMARY KEY AUTO_INCREMENT, author_id INTEGER UNSIGNED NOT NULL, CONSTRAINT fk_author_book FOREIGN KEY Book(author_id) REFERENCES Author(id) ON DELETE CASCADE ON UPDATE NO ACTION, name VARCHAR(255) NOT NULL);',
         ];
 
         yield 'postgresql' => [
-            new PostgreSqlDriver(),
+            DatabaseDialect::POSTGRESQL,
             'CREATE TABLE Book (id SERIAL PRIMARY KEY, author_id INTEGER UNSIGNED NOT NULL, CONSTRAINT fk_author_book FOREIGN KEY Book(author_id) REFERENCES Author(id) ON DELETE CASCADE ON UPDATE NO ACTION, name VARCHAR(255) NOT NULL);',
         ];
 
         yield 'sqlite' => [
-            new SQLiteDriver(),
+            DatabaseDialect::SQLITE,
             'CREATE TABLE Book (id INTEGER PRIMARY KEY AUTOINCREMENT, author_id INTEGER UNSIGNED NOT NULL, name VARCHAR(255) NOT NULL);',
         ];
     }
 
-    public static function provide_database_driver(): Generator
-    {
-        yield 'mysql' => [
-            new MySqlDriver(),
-        ];
-
-        yield 'postgresql' => [
-            new PostgreSqlDriver(),
-        ];
-
-        yield 'sqlite' => [
-            new SQLiteDriver(),
-        ];
-    }
-
     #[Test]
     #[DataProvider('provide_alter_table_syntax')]
-    public function it_can_create_an_alter_table_add_statement(DatabaseDriver $driver, string $operation, string $validSql): void
+    public function it_can_create_an_alter_table_add_statement(DatabaseDialect $dialect, string $operation, string $validSql): void
     {
-        $this->markTestIncomplete('TODO: add alter table statement');
-        //        $statement = $driver->dialect()->createQueryStatement('Author')
-        //            ->alterTable($operation)
-        //            ->createColumn('name', 'VARCHAR(255)');
-        //
-        //        $this->assertSame($validSql, (string) $statement);
+        $query = (new AlterTableStatement('Author'))
+            ->$operation(new VarcharStatement('name'))
```

Comment:
this might be a bit magical, but it allows us to use the referenced method from the dataprovider

Reply:
It's ok :)